### PR TITLE
feat: validate that password is not equal to email

### PR DESCRIPTION
**Pull request by timurgilfanov**
Closes #328

## Summary
Adds a cross-field credentials validation rule that rejects passwords matching the entered email address, including the user-facing error string and localized German resource.

The equality check remains inside `CredentialsValidatorImpl` so credentials validation stays centralized in the feature domain validator. The check is performed after the existing password rules, preserving the current validation ordering while adding the cross-field rule requested by the issue.

## Changes
- Added `PasswordValidationError.PasswordEqualToEmail` and mapped it to auth display strings.
- Updated `CredentialsValidatorImpl` to run the password-equals-email check after existing password validation succeeds.
- Added and updated auth validator and real-time ViewModel tests for the new validation path.

## Testing
Validation recorded in the completed plan:
- [x] Unit tests added or updated
- [ ] Manually tested
- [x] Edge cases considered

Commands recorded as run:
- `./gradlew :feature:auth:testDebugUnitTest --tests "timur.gilfanov.messenger.auth.domain.validation.CredentialsValidatorImplTest"`

### CODE REVIEW

File: feature/auth/src/main/kotlin/timur/gilfanov/messenger/auth/ui/screen/login/LoginViewModel.kt [hunk 1]
```diff
@@ -58,30 +59,51 @@ class LoginViewModel @Inject constructor(
...
+        )
+        val isCredentialsValid = credentialsResult is ResultWithError.Success
+        val credError = (credentialsResult as? ResultWithError.Failure)?.error
+        val passwordError = when (credError) {
+            is CredentialsValidationError.Password -> credError.reason
+            else -> (
+                credentialsValidator.validate(
+                    Password(password),
+                ) as? ResultWithError.Failure
+                )?.error
```
Comment (timurgilfanov) [inline feature/auth/src/main/kotlin/timur/gilfanov/messenger/auth/ui/screen/login/LoginViewModel.kt:107]: Why we need to validate password here?

File: feature/auth/src/main/kotlin/timur/gilfanov/messenger/auth/ui/screen/login/LoginViewModel.kt [hunk 2]
```diff
@@ -58,30 +59,51 @@ class LoginViewModel @Inject constructor(
...
+        val passwordError = when {
+            currentPassword.isEmpty() -> _state.value.passwordError
+            credError is CredentialsValidationError.Password -> credError.reason
+            credentialsResult is ResultWithError.Success -> null
+            else -> (
+                credentialsValidator.validate(
+                    Password(currentPassword),
+                ) as? ResultWithError.Failure
+                )?.error
+        }
```
Comment (timurgilfanov) [inline feature/auth/src/main/kotlin/timur/gilfanov/messenger/auth/ui/screen/login/LoginViewModel.kt:82]: Nice to add an explanation comment for this complex expression.

File: feature/auth/src/main/kotlin/timur/gilfanov/messenger/auth/ui/screen/login/LoginViewModel.kt (hunk 1 above)
Comment (timurgilfanov) [reply feature/auth/src/main/kotlin/timur/gilfanov/messenger/auth/ui/screen/login/LoginViewModel.kt:107]: To set password error in case there is active email error because email check have a priority in credentials validation in `CredentialValidatorImpl`.

File: feature/auth/src/main/kotlin/timur/gilfanov/messenger/auth/ui/screen/login/LoginViewModel.kt (hunk 2 above)
Comment (timurgilfanov) [reply feature/auth/src/main/kotlin/timur/gilfanov/messenger/auth/ui/screen/login/LoginViewModel.kt:82]: ok
